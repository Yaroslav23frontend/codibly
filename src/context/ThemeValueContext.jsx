import React, { useContext, useState } from 'react';

const ThemeValueContext = React.createContext();

export function useThemeValue() {
  return useContext(ThemeValueContext);
}
export default function ThemeValueProivider({ children }) {
  const [mode, setMode] = useState(
    localStorage.getItem('mode') ? localStorage.getItem('mode') : 'dark'
  );
  const value = {
    mode,
    setMode
  };
  return (
    <ThemeValueContext.Provider value={value}>
      {children}
    </ThemeValueContext.Provider>
  );
}
