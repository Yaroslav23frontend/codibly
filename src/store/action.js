export const uploadTable = "UPLOAD_TABLE";
