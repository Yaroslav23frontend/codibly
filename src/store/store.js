import { configureStore } from '@reduxjs/toolkit';
import { reducerTable } from './reducers/reducerTable.ts';

export const store = configureStore({
  reducer: {
    table: reducerTable
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false
    })
});
