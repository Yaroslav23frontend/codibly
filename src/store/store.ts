import { configureStore } from '@reduxjs/toolkit';
import { reducerTable } from './reducers/reducerTable.ts';
type TableFormat = {
  id: number,
  name: string,
  color: string,
  year: string
}
export interface IRootState {
  table: Array<TableFormat>
}
export const store = configureStore({
  reducer: {
    table: reducerTable
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false
    })
});
