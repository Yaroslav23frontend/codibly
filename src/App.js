import { ThemeProvider, createTheme } from "@mui/material/styles";
import { BrowserRouter } from "react-router-dom";
import AppRouter from "./components/AppRouter";
import { useThemeValue } from "./context/ThemeValueContext";
import { useMemo } from "react";
function App() {
  const { mode } = useThemeValue();
  const theme = useMemo(() => {
    const theme = createTheme({
      palette: {
        mode: mode,
      },
    });
    return theme;
  }, [mode]);
  return (
    <BrowserRouter>
      <ThemeProvider theme={theme}>
        <AppRouter />
      </ThemeProvider>
    </BrowserRouter>
  );
}

export default App;
