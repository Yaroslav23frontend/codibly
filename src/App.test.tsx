import { render, screen, waitFor, fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from './store/store';
import ThemeValueProivider from './context/ThemeValueContext';
import App from './App';
import userEvent from '@testing-library/user-event';
import { act, renderHook } from '@testing-library/react-hooks';
import axios, { AxiosResponse } from 'axios';
import { useThemeValue } from './context/ThemeValueContext';
describe('Table test', () => {
  afterEach(() => {
    jest.restoreAllMocks();
  });
  const mAxiosResponse: object = {
    data: {
      page: 1,
      per_page: 5,
      total: 12,
      total_pages: 3,
      data: [
        {
          id: 1,
          name: 'cerulean',
          year: 2000,
          color: '#98B2D1',
          pantone_value: '15-4020'
        },
        {
          id: 2,
          name: 'fuchsia rose',
          year: 2001,
          color: '#C74375',
          pantone_value: '17-2031'
        },
        {
          id: 3,
          name: 'true red',
          year: 2002,
          color: '#BF1932',
          pantone_value: '19-1664'
        },
        {
          id: 4,
          name: 'aqua sky',
          year: 2003,
          color: '#7BC4C4',
          pantone_value: '14-4811'
        },
        {
          id: 5,
          name: 'tigerlily',
          year: 2004,
          color: '#E2583E',
          pantone_value: '17-1456'
        }
      ],
      support: {
        url: 'https://reqres.in/#support-heading',
        text: 'To keep ReqRes free, contributions towards server costs are appreciated!'
      }
    }
  } as AxiosResponse;
  test('buttons: next, back, first and last', async () => {
    jest.spyOn(axios, 'get').mockResolvedValueOnce(mAxiosResponse);

    act(() => {
      render(
        <Provider store={store}>
          <ThemeValueProivider>
            <App />
          </ThemeValueProivider>
        </Provider>
      );
    });

    const page = await screen.findByTestId('page');
    const next = await screen.getByTestId('next-button');
    const back = await screen.getByTestId('back-button');
    const last = await screen.getByTestId('last-button');
    const first = await screen.getByTestId('first-button');
    await waitFor(() => userEvent.click(next));
    expect(Number(page.textContent)).toBe(2);
    await waitFor(() => userEvent.click(back));
    expect(Number(page.textContent)).toBe(1);
    await waitFor(() => userEvent.click(last));
    expect(Number(page.textContent)).toBe(3);
    await waitFor(() => userEvent.click(first));
    expect(Number(page.textContent)).toBe(1);
  });

  test('filter', async () => {
    jest.spyOn(axios, 'get').mockResolvedValueOnce(mAxiosResponse);
    act(() => {
      render(
        <Provider store={store}>
          <ThemeValueProivider>
            <App />
          </ThemeValueProivider>
        </Provider>
      );
    });
    const input = (await screen.getByLabelText(
      /Filter by id/i
    )) as HTMLInputElement;
    const items = await screen.findAllByTestId('table-item');
    fireEvent.change(input, { target: { value: 1 } });
    console.log(input.value);
    expect(input.value).toBe('1');
    expect(items.length).toBeGreaterThan(1);
  });
  test('theme', async () => {
    jest.spyOn(axios, 'get').mockResolvedValueOnce(mAxiosResponse);
    const wrapper = ({ children }) => (
      <ThemeValueProivider>{children}</ThemeValueProivider>
    );
    const mode = renderHook(() => useThemeValue(), { wrapper });
    act(() => {
      render(
        <Provider store={store}>
          <ThemeValueProivider>
            <App />
          </ThemeValueProivider>
        </Provider>
      );
    });
    const themebutton = await screen.getByTestId('theme-button');
    userEvent.click(themebutton);
    waitFor(() => expect(mode.result.current.mode).toBe('ligth'));
    userEvent.click(themebutton);
    waitFor(() => expect(mode.result.current.mode).toBe('dark'));
  });
});
