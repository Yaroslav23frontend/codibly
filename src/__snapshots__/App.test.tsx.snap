// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table test next page 2 1`] = `
<p
  class="MuiTypography-root MuiTypography-body1 css-1k9rmp9-MuiTypography-root"
  data-testid="page"
>
  1
</p>
`;
